deepmd/nvnmd/entrypoints/freeze.py

```diff
@@ -0,0 +1,307 @@
+    """Get node names based on model type.
+
+    Parameters
+    ----------
+    model_type : str
+        str type of model
+    modifier_type : Optional[str], optional
+        modifier type if any, by default None
+
+    Returns
+    -------
+    List[str]
+        list with all node names to freeze
+
+    Raises
+    ------
+    RuntimeError
+        if unknown model type
+    """
+    nodes = [
+        "model_type",
+        "descrpt_attr/rcut",
+        "descrpt_attr/ntypes",
+        "model_attr/tmap",
+        "model_attr/model_type",
+        "model_attr/model_version",
+        "train_attr/min_nbor_dist",
+        "train_attr/training_script",
+    ]
+
+    if model_type == "ener":
+        nodes += [
+            "o_energy",
+            "o_force",
+            "o_virial",
+            "o_atom_energy",
+            "o_atom_virial",
+            "fitting_attr/dfparam",
+            "fitting_attr/daparam",
+        ]
+    elif model_type == "wfc":
+        nodes += [
+            "o_wfc",
+            "model_attr/sel_type",
+            "model_attr/output_dim",
+        ]
+    elif model_type == "dipole":
+        nodes += [
+            "o_dipole",
+            "o_global_dipole",
+            "o_force",
+            "o_virial",
+            "o_atom_virial",
+            "o_rmat",
+            "o_rmat_deriv",
+            "o_nlist",
+            "o_rij",
+            "descrpt_attr/sel",
+            "descrpt_attr/ndescrpt",
+            "model_attr/sel_type",
+            "model_attr/output_dim",
+        ]
+    elif model_type == "polar":
+        nodes += [
+            "o_polar",
+            "o_global_polar",
+            "o_force",
+            "o_virial",
+            "o_atom_virial",
+            "model_attr/sel_type",
+            "model_attr/output_dim",
+        ]
+    elif model_type == "global_polar":
+        nodes += [
+            "o_global_polar",
+            "model_attr/sel_type",
+            "model_attr/output_dim",
+        ]
+    else:
+        raise RuntimeError(f"unknow model type {model_type}")
+    if modifier_type == "dipole_charge":
+        nodes += [
+            "modifier_attr/type",
+            "modifier_attr/mdl_name",
+            "modifier_attr/mdl_charge_map",
+            "modifier_attr/sys_charge_map",
+            "modifier_attr/ewald_h",
+            "modifier_attr/ewald_beta",
+            "dipole_charge/model_type",
+            "dipole_charge/descrpt_attr/rcut",
+            "dipole_charge/descrpt_attr/ntypes",
+            "dipole_charge/model_attr/tmap",
+            "dipole_charge/model_attr/model_type",
+            "dipole_charge/model_attr/model_version",
+            "o_dm_force",
+            "dipole_charge/model_attr/sel_type",
+            "dipole_charge/o_dipole",
+            "dipole_charge/model_attr/output_dim",
+            "o_dm_virial",
+            "o_dm_av",
+        ]
+    return nodes
+
+
+def freeze_nvnmd(
```

Comment:
The `freeze_nvnmd` method is almost the same as the standard `freeze`. How about merging them ?